src/vmprof_common.h

```diff
@@ -32,6 +35,42 @@ typedef struct prof_stacktrace_s {
     void *stack[];
 } prof_stacktrace_s;
 
+#if !DISABLE_GZIP
+static int pipe_gzip(int out_fd, int *gzip_out_fd, int *subproc_pid)
+{
+    int pipefds[2];
+
+    if (pipe(pipefds) == -1)
+        return -1;
+
+    pid_t pid = fork();
+    switch (pid) {
+    case -1:
+        close(pipefds[0]);
+        close(pipefds[1]);
+        return -1;
+    case 0:
+        dup2(pipefds[0], STDIN_FILENO);
+        dup2(out_fd, STDOUT_FILENO);
+        close(pipefds[0]);
+        close(pipefds[1]);
+        close(out_fd);
+        /* Try system gzip */
+        execlp("gzip", "gzip", NULL);
+        perror("gzip");
+        /* Fall back to Python gzip module */
+        execlp("python", "python", "-m", "gzip", NULL);
+        perror("python -m gzip");
+        /* TODO: How do we avoid "hang" in the parent process in this case? */
```

Comment:
Also if you put something stupid in here so that the child process doesn't start, the parent process hangs... which I don't understand, shouldn't it get a broken pipe?